testsuite/MDAnalysisTests/analysis/test_hbonds.py

```diff
@@ -193,9 +192,8 @@ def _setUp(self):
     def _run(self, **kwargs):
         kw = kwargs.copy()
         kw.update(kwargs)
-        with warnings.catch_warnings():
-            # ignore SelectionWarning
-            warnings.simplefilter("ignore")
+        # ignore UserWarning
+        with pytest.warns(UserWarning):
```

Comment:
@richardjgowers , @orbeckst 

This change works under python 3.x but breaks python 2.7. This means we have different behavior here. I do not know enough about the hbond code to check what might be a problem here.

Reply:
OK so under python3 we **always** raise a `UserWarning`. We never do that under python 2.7. But under python 2.7 we occasionally raise `SelectionWarning`.